src/barfTurn.ts

```diff
@@ -338,6 +367,34 @@ const turns: AdventureAction[] = [
     spendsTurn: false,
     sobriety: Sobriety.SOBER,
   },
+  {
+    name: "Pig Skinner Free-For-All",
+    available: () =>
+      have($skill`Free-For-All`) &&
+      !have($effect`Everything Looks Red`) &&
+      romanticMonsterImpossible(),
+    execute: () => {
+      const usingDuplicate = SourceTerminal.have() && SourceTerminal.duplicateUsesRemaining() > 0;
+      const location = wanderWhere("pigskinner");
+      const familiar = freeFightFamiliar({ location, allowAttackFamiliars: !usingDuplicate });
```

Comment:
I'm not sure if we currently have code to prevent the use of the crimbo tree, but we should prevent the use of the crimbo tree

Reply:
What crimbo tree? Why?

Reply:
shrubling

Reply:
This looks covered already [where we select familiars](https://github.com/Loathing-Associates-Scripting-Society/garbage-collector/blob/main/src/familiar/freeFightFamiliar.ts#L53) and [how we check if we can open a red present](https://github.com/Loathing-Associates-Scripting-Society/garbage-collector/blob/37a1bffa82600acbd569a7d5ddbca7ca8846e4ff/src/familiar/lib.ts#L77)

Reply:
Oh, but the issue is we can try to do both together?

Reply:
But any time we're eligible for Free For All, we necessarily don't have ELR, which means we're eligible to open a present; we need to add a condition to `canOpenRedPresent` that says that if we have Free For All, we can't open a present